scripts/common.sh

```diff
@@ -135,3 +138,26 @@ function extract_debs_into_dir() {
 	find "$source_dir" -name '*.deb' -exec mv {} "$target_dir" \;
 	find "$source_dir" -name '*.ddeb' -exec mv {} "$target_dir" \;
 }
+
+function get_upstream_or_fail_if_unset() {
+	if [[ -z "$UPSTREAM_PRODUCT_BRANCH" ]]; then
+		local upstream_branch
+		upstream_branch="$(git rev-parse --abbrev-ref --symbolic-full-name "@{u}" | cut -d'/' -f2-)"
```

Comment:
Does this work in detached HEAD ? Jenkins  GitSCM checkouts result in a detached HEAD on the cloned repo so I think this won't work. I think it would fail with `HEAD: does not point to a branch`

Not sure if this is needed or if on Jenkins it will always be set explicitly via UPSTREAM_PRODUCT_BRANCH so we don't have to determine it. But thought I would mention it.